### PR TITLE
Bump rust to 1.74

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
- Use new `lint` table.

  https://doc.rust-lang.org/stable/cargo/reference/manifest.html#the-lints-section
  https://doc.rust-lang.org/stable/cargo/reference/workspaces.html#the-lints-table

- Fix const ref without an explicit static lifetime.
- Remove unnecessary `let` binding for downcast.
- Fix the borrowed expression implements the required traits for `std::fs::read_dir(path)`.
- Fix redundant redifinition of a binding.
- Ignore warning about usage of `skip(0)`.
- Ignore unwrap used in example `memfs`.

### CODE REVIEW

File: libparsec/crates/platform_mountpoint/examples/memfs.rs [hunk 1]
```diff
@@ -16,10 +16,13 @@ fn main() {
         let arg = args.nth(1).expect("Please provide mountpoint path");
         let path = Path::new(&arg);
 
+        #[allow(clippy::unwrap_used)]
```
Comment (TimeEngineer) [inline libparsec/crates/platform_mountpoint/examples/memfs.rs:19]: Can you use global allow ?

**Comment by TimeEngineer** [comment]
> Fix usage of skip(0) use 1 to skip the first element

What do you mean ? You fixed with a `clippy allow`

**Comment by TimeEngineer** [comment]
> Fix & without an explicit lifetime.

This is for const only ? so maybe you mean without explicit static lifetime ?

File: libparsec/crates/platform_mountpoint/examples/memfs.rs (hunk 1 above)
Comment (FirelightFlagboy) [reply libparsec/crates/platform_mountpoint/examples/memfs.rs]: done

**Comment by FirelightFlagboy** [comment]
@TimeEngineer I've updated the commit message (I often forget to update the PR description)

**Review by TimeEngineer — APPROVED** [review]

**Review by Max-7 — APPROVED** [review]

**Review by mmmarcos — APPROVED** [review]

**Pull request merged**